src/load_balancer.rs

```diff
@@ -0,0 +1,70 @@
+use std::{str::FromStr, sync::Arc};
+
+use hyper::{Body, Client, Request, Uri};
+use tokio::sync::RwLock;
+
+use crate::strategies::LBStrategy;
+
+pub struct LoadBalancer {
+    client: Client<hyper::client::HttpConnector>,
+    strategy: Arc<RwLock<dyn LBStrategy + Send + Sync>>,
```

Comment:
You could create a type for this line of code for easy readability and usage in all other places `Arc<RwLock<dyn LBStrategy + Send + Sync>> ` would appear